core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -836,13 +836,22 @@ protected void onException(Channel channel, Exception e) throws IOException {
         } else if (e instanceof TcpTransport.HttpOnTransportException) {
             // in case we are able to return data, serialize the exception content and sent it back to the client
             if (isOpen(channel)) {
-                sendMessage(channel, new BytesArray(e.getMessage().getBytes(StandardCharsets.UTF_8)), () -> {
+                final Runnable closeChannel = () -> {
                     try {
                         closeChannels(Collections.singletonList(channel));
                     } catch (IOException e1) {
                         logger.debug("failed to close httpOnTransport channel", e1);
                     }
-                });
+                };
+                boolean success = false;
+                try {
+                    sendMessage(channel, new BytesArray(e.getMessage().getBytes(StandardCharsets.UTF_8)), closeChannel);
+                    success = true;
+                } finally {
+                    if (success == false) {
+                        closeChannel.run();
```

Comment:
nit - can we add a line to the java docs of closeChannels that it can be called on already closed channels?